### PR TITLE
XS-3026 Implement DQC 1

### PR DESCRIPTION
**Pull request by mindeerickson-wf**
Original PR: https://github.com/DataQualityCommittee/dqc_us_rules/pull/104

Changes:

Added rules and config for DQC 1
Added unit tests.
Please review: @hefischer @andrewperkins-wf

### CODE REVIEW

**Comment by chrislococo-wf** [comment]
+1

File: dqc_us_rules/dqc_us_0001.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+                if fact is not None:
+                    val.modelXbrl.error(
+                        '{base_key}.{extension_key}'.format(
+                            base_key=_CODE_NAME,
+                            extension_key=axis_config[_RULE_INDEX_KEY]
+                        ),
+                        messages.get_message(_CODE_NAME, _UGT_FACT_KEY),
+                        axis=axis.label(),
+                        member=child.label(),
+                        modelObject=fact,
```
Comment (brianmyers-wf) [inline dqc_us_rules/dqc_us_0001.py]: hate this if else setup, but not seeing a better way to do it... do you know if you can send in a none as a parameter? if so maybe something like this?
```
val.modelXbrl.error(
    '{base_key}.{extension_key}'.format(
        base_key=_CODE_NAME,
        extension_key=axis_config[_RULE_INDEX_KEY]
    ),
    messages.get_message(_CODE_NAME, _UGT_FACT_KEY),
    axis=axis.label(),
    member=child.label(),
    ruleVersion=_RULE_VERSION,
    modelObject=fact,
    group=role if fact is None else None
```

File: dqc_us_rules/dqc_us_0001.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    return (
+        isinstance(concept, ModelConcept) and
+        concept.qname
+    )
+
+
+def _all_members_under(axis, relset):
+    """
+    Returns a dictionary of concepts seen under the provided concept,
+    in the given relset and filtered by the optional filter.
```
Comment (brianmyers-wf) [inline dqc_us_rules/dqc_us_0001.py]: what "optional filter"?

File: dqc_us_rules/util/facts.py [hunk 3]
```diff
@@ -331,6 +331,33 @@ def legal_entity_axis_facts_by_member(facts):
...
+                dim for dim in fact.context.segDimValues.values()
+                if (
+                    dim.isExplicit and
+                    dim.member is not None and
+                    dim.member.qname.localName == member_name
+                )
+            ]
+            for dim in dims:
+                if dim.dimension.qname.localName == axis_name:
+                    return fact
```
Comment (brianmyers-wf) [inline dqc_us_rules/util/facts.py]: can there only be one fact? seems like you're only returning the first found, but couldn't there potentially be many?

**Comment by brianmyers-wf** [comment]
pass complete

File: dqc_us_rules/util/facts.py (hunk 3 above)
Comment (jasonleinbach-wf) [inline dqc_us_rules/util/facts.py]: I lifted this from wf_arelle axis modeling stuff.  I'm guessing that there can only be one fact.

File: dqc_us_rules/dqc_us_0001.py (hunk 2 above)
Comment (jasonleinbach-wf) [inline dqc_us_rules/dqc_us_0001.py]: I removed the filter part and forgot to update the docstring.

File: dqc_us_rules/dqc_us_0001.py (hunk 1 above)
Comment (jasonleinbach-wf) [inline dqc_us_rules/dqc_us_0001.py]: It takes different messages, as well as a group parameter.  I'd prefer to go with verbose but simple.

**Comment by brianmyers-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
+1

**Comment by brianmyers-wf** [comment]
QA +1
Testing notes can be found in the Jira ticket.

**Comment by andrewperkins-wf** [comment]
reviewed and tested.  Merging.

**Pull request merged**